vision_agent/tools/tools.py

```diff
@@ -643,6 +639,58 @@ def __call__(self, mask1: Union[str, Path], mask2: Union[str, Path]) -> float:
         return cast(float, round(iou, 2))
 
 
+class BboxContains(Tool):
+    name = "bbox_contains_"
+    description = "Given two bounding boxes, a target bounding box and a region bounding box, 'bbox_contains_' returns the intersection of the two bounding boxes over the target bounding box, reflects the percentage area of the target bounding box overlaps with the region bounding box. This is a good tool for determining if the region object contains the target object."
```

Comment:
Just curious, did you test any other workflows with this? I've noticed that the model can have a tendency to pick tools with longer descriptions, even if the tool is not the best fit. So for example, if you test another workflow where you want it to use a similar tool like BboxIoU and see if it gets it or mixes it up.

Reply:
tested two cases: the coffee jar and two poker cards
prompts:
“What percentage of this jar is filled up with coffee beans?” it used seg_area
“Do these two cards overlap?“, it uses seg_iou in the first try, and bbox_iou after reflection